lib/internal/test_runner/fake_timers.js

```diff
@@ -0,0 +1,110 @@
+'use strict';
+
+const {
+  DateNow,
+  SafeMap,
+  Symbol,
+  globalThis,
+} = primordials;
+
+class Timers {
+  constructor() {
+    this.timers = new SafeMap();
+
+    this.setTimeout = this.#createTimer.bind(this, false);
+    this.clearTimeout = this.#clearTimer.bind(this);
+    this.setInterval = this.#createTimer.bind(this, true);
+    this.clearInterval = this.#clearTimer.bind(this);
+  }
+
+  #createTimer(isInterval, callback, delay, ...args) {
+    const timerId = Symbol('kTimerId');
+    const timer = {
+      id: timerId,
+      callback,
+      time: DateNow() + delay,
+      interval: isInterval,
+      args,
+    };
+    this.timers.set(timerId, timer);
+    return timerId;
+  }
+
+  #clearTimer(timerId) {
+    this.timers.delete(timerId);
+  }
+
+}
+
+let realSetTimeout;
+let realClearTimeout;
+let realSetInterval;
+let realClearInterval;
+
+class FakeTimers {
+  constructor() {
+    this.fakeTimers = {};
+    this.isEnabled = false;
+    this.now = DateNow();
+  }
+
+  tick(time = 0) {
+
+    // if (!this.isEnabled) {
+    //   throw new Error('you should enable fakeTimers first by calling the .enable function');
+    // }
+
+    this.now += time;
+    const timers = this.fakeTimers.timers;
+
+    for (const timer of timers.values()) {
+
+      if (!(this.now >= timer.time)) continue;
+
+      timer.callback(...timer.args);
+      if (timer.interval) {
+        timer.time = this.now + (timer.time - this.now) % timer.args[0];
+        continue;
+      }
+
+      timers.delete(timer.id);
+    }
+  }
+
+  enable() {
+    // if (this.isEnabled) {
+    //   throw new Error('fakeTimers is already enabled!');
+    // }
+    this.now = DateNow();
+    this.isEnabled = true;
+    this.fakeTimers = new Timers();
+
+    realSetTimeout = globalThis.setTimeout;
+    realClearTimeout = globalThis.clearTimeout;
+    realSetInterval = globalThis.setInterval;
+    realClearInterval = globalThis.clearInterval;
+
+    globalThis.setTimeout = this.fakeTimers.setTimeout;
```

Comment:
In order for this feature to be useful a lot of other stuff needs to be mocked namely:
 - Date.now/performance.now
 - timers/promises
 - scheduler.wait 
 - queueMicrotask/process.nextTick (maybe?)


And if this feature is in core ideally stuff like HTTP date headers and such